floodlight/core/pitch.py

```diff
@@ -147,3 +153,967 @@ def center(self):
+            )
+
+        # initialize essential parameters for plotting with matplotlib
+        scalex = kwargs.get("scalex", False)
+        scaley = kwargs.get("scaley", False)
+
+        # check wether an axes to plot is given or if a new axes element
+        # has to be created
+        ax = ax or plt.subplots(**kwargs)[1]
+        ax.set_aspect(1)  # ensures that the ratio between length and width is correct
+        # regardless of the figsize
+
+        # create axes with handball pitch
+        if sport == "handball":
+            return self._plot_handball_pitch(
+                scalex=scalex,
+                scaley=scaley,
+                color_scheme=color_scheme,
+                save=save,
+                ax=ax,
+                **kwargs,
+            )
+
+        # create axes with soccer/football pitch
+        if sport == "soccer" or "football":
+            return self._plot_soccer_pitch(
+                scalex=scalex,
+                scaley=scaley,
+                color_scheme=color_scheme,
+                save=save,
+                ax=ax,
+                **kwargs,
+            )
+
+    def _plot_handball_pitch(
+        self,
+        scalex: bool,
+        scaley: bool,
+        color_scheme: str,
+        save: bool,
+        ax: plt.axes,
+        **kwargs,
+    ):
+
+        """Plots a handball pitch on a given axes
+
+        Parameters
+        ----------
+        scalex: bool
+            Determines if the view limits are adapted to the
+            data limits of the x values.
+        scaley: bool
+            Determines if the view limits are adapted to the
+            data limits of the y values.
+        color_scheme: str
+             Color scheme of the plot. One of {'normal', 'black_white'}.
+             If not given 'normal' is the default color scheme.
+        save: bool
+            You may want to save the graph. If the graphic should be saved 'save'
+            must be True. If not given as an argument the pitch is not going to be
+            saved. Plots are going to be stored as .png on the same directory layer as
+            the repository.
+        kwargs:
+            You may pass optional arguments (`linewidth`} which can be used for the plot
+            with matplotlib.
+        ax: plt.axes
+            Axes from matplotlib library on which the handball field is plotted.
+
+        Returns
+        ----------
+        ax : matplotlib.axes._subplots.AxesSubplot
+            An axes to which all elements of the sport specific pitch are added.
+        """
+
+        # customizing visualization
+        if color_scheme == "black_white":
+            ax.set_facecolor("white")
+            outline = "black"
+            innerline = "black"
+            fill = "white"
+            goal = "black"
+        else:
+            ax.set_facecolor("skyblue")
+            outline = "white"
+            innerline = "white"
+            fill = "khaki"
+            goal = "white"
```

Comment:
I think more descriptive variable names would be helpful here by prefixing everything with ``color`` and referring to the pitch part they belong to (what is being filled?), e.g.
 ``color_contour_lines``, ``color_inner_lines``, ``color_penalty_area_fill?``, ``color_goal`` ?

Reply:
Renamed.